.github/workflows/dev.yaml

```diff
@@ -97,4 +99,16 @@ jobs:
             # This generates a signature and publishes it to the registry, next to the image
             # Uses the keyless signing flow with Github Actions as identity provider
             cosign sign -y "$IMAGE_NAME@$DIGEST"
+
+            # Generate SBOM for the image
+            syft scan --output cyclonedx-json=sbom.json --select-catalogers "-cargo-auditable-binary-cataloger" --scope all-layers --source-name "${{ matrix.product }}" --source-version "$TAG_NAME" "$IMAGE_NAME@$DIGEST";
+            # Determine the PURL for the image
+            PURL="pkg:docker/sdp/${{ matrix.product }}@$DIGEST?repository_url=oci.stackable.tech";
+            # Get metadata from the image
+            IMAGE_METADATA_DESCRIPTION=$(docker inspect --format='{{.Config.Labels.description}}' "$IMAGE_NAME@$DIGEST");
+            IMAGE_METADATA_NAME=$(docker inspect --format='{{.Config.Labels.name}}' "$IMAGE_NAME@$DIGEST");
+            # Merge the SBOM with the metadata for the image
+            jq -s '{"metadata":{"component":{"description":"'"$IMAGE_METADATA_NAME. $IMAGE_METADATA_DESCRIPTION"'","supplier":{"name":"Stackable GmbH","url":["https://stackable.tech/"]},"author":"Stackable GmbH","purl":"'"$PURL"'","publisher":"Stackable GmbH"}}} * .[0]' sbom.json > sbom.merged.json;
+            # Attest the SBOM to the image
+            cosign attest -y --predicate sbom.merged.json --type cyclonedx "$IMAGE_NAME@$DIGEST"
```

Comment:
These inline scripts are getting rather big. I wonder if we should pull them out into their own files, or make [reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows), especially since some parts are duplicated across jobs?

On another note, Github Actions allows [grouping log lines](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#grouping-log-lines) (Unrelated, but the [Job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary) is cool, I hadn't known about that).

Reply:
_I have added an Architecture Meeting agenda item. Feel free to mark as resolved once you've read it._

Reply:
Please see this comment/thread: https://github.com/stackabletech/operator-templating/pull/312#discussion_r1466376618